packages/medusa/src/models/batch-job.ts

```diff
@@ -18,10 +19,32 @@ export class BatchJob extends SoftDeletableEntity {
   created_by_user: User
 
   @DbAwareColumn({ type: "jsonb", nullable: true })
-  context: { retry_count?: number; max_retry?: number } & Record<string, unknown>
+  context: {
+    retry_count?: number
+    max_retry?: number
+    offset?: number
+    limit?: number
+    order?: string
+    fields?: string
+    expand?: string
+    file_key?: string
+    list_config?: {
+      select?: string[]
+      relations?: string[]
+      skip?: number
+      take?: number
+      order?: Record<string, "ASC" | "DESC">
+    }
+    filterable_fields?: Selector<unknown>
```

Comment:
thought: I don't think this should be reflected in the entity type. The shape of the context field is a contractual agreement between the batch job strategy and itself.

I would leave the `context` type very open and not force/imply a certain shape on the BatchJobStrategy author. If it becomes annoying to work with you can maybe cast the type within the BatchJobStrategy itself, but wouldn't want this to be globally set.

Reply:
Yes, indeed i created a Context type and then moved it it. I will move it back to the strategy level, except for the fileKey that should be the same no matter the strategy to be sure to find it easily. Wdyt?